portfolio-project-matching-app/models/Technology.js

```diff
@@ -1,15 +1,54 @@
+class Technology {
+    // fields
+    id;
+    name;
+    language;
+    backend;
 
+    // static factory methods (used like overloaded constructors)
+    static fromDocSnapshot(id, docSnapshot) {
```

Comment:
I added some static factory methods in place of having a single constructor. This gives the developer a bit more freedom when creating one of our custom objects / classes.